.circleci/config.yml

```diff
@@ -1,193 +1,334 @@
+    environment: *global_environment_vars
     steps:
       - setup_remote_docker
+      - checkout
       - <<: *attach_workspace
       - <<: *load_docker_image
+      - <<: *ssh_forward_port
       - run:
-          name: Push to Dockerhub
-          command: |
-            echo "$DOCKER_PASS" | docker login -u "$DOCKER_USER" --password-stdin
-            export DEV_IMG_TAG="anchore/anchore-cli-dev:${CIRCLE_SHA1}"
-            docker tag "${IMAGE_NAME}:ci" "$DEV_IMG_TAG"
-            docker push "$DEV_IMG_TAG"
-            if [ "$CIRCLE_BRANCH" == "master" ]; then
-              echo "${IMAGE_NAME}:${IMAGE_TAG}"
-              docker tag "${IMAGE_NAME}:ci" "${IMAGE_NAME}:${IMAGE_TAG}"
-              docker push "${IMAGE_NAME}:${IMAGE_TAG}"
-            elif [ "$CIRCLE_BRANCH" == "0.2" ]; then
-              echo "${IMAGE_NAME}:v0.2-${IMAGE_TAG}"
-              docker tag "${IMAGE_NAME}:ci" "${IMAGE_NAME}:v0.2-${IMAGE_TAG}"
-              docker push "${IMAGE_NAME}:v0.2-${IMAGE_TAG}"
-            elif [ "$CIRCLE_BRANCH" == "0.3" ]; then
-              echo "${IMAGE_NAME}:v0.3-${IMAGE_TAG}"
-              docker tag "${IMAGE_NAME}:ci" "${IMAGE_NAME}:v0.3-${IMAGE_TAG}"
-              docker push "${IMAGE_NAME}:v0.3-${IMAGE_TAG}"
-            elif [ "$CIRCLE_BRANCH" == "0.4" ]; then
-              echo "${IMAGE_NAME}:v0.4-${IMAGE_TAG}"
-              docker tag "${IMAGE_NAME}:ci" "${IMAGE_NAME}:v0.4-${IMAGE_TAG}"
-              docker push "${IMAGE_NAME}:v0.4-${IMAGE_TAG}"
-            elif [ "$CIRCLE_BRANCH" == "0.5" ]; then
-              echo "${IMAGE_NAME}:v0.5-${IMAGE_TAG}"
-              docker tag "${IMAGE_NAME}:ci" "${IMAGE_NAME}:v0.5-${IMAGE_TAG}"
-              docker push "${IMAGE_NAME}:v0.5-${IMAGE_TAG}"
-            elif [ "$CIRCLE_BRANCH" == "0.6" ]; then
-              echo "${IMAGE_NAME}:v0.6-${IMAGE_TAG}"
-              docker tag "${IMAGE_NAME}:ci" "${IMAGE_NAME}:v0.6-${IMAGE_TAG}"
-              docker push "${IMAGE_NAME}:v0.6-${IMAGE_TAG}"
-            fi
-
-  trigger_downstream:
-      docker:
-        - image: anchore/test-infra:latest
-          auth:
-            username: $DOCKER_USER
-            password: $DOCKER_PASS
-      steps:
+          name: Run e2e tests
+          command: make test-e2e
+
+  push_image:
+    description: Uses make script to push images to DockerHub - dev, rc, prod
+    parameters:
+      make_job:
+        type: string
+    docker:
+      - image: circleci/python:3.6
+    environment: *global_environment_vars
+    steps:
+      - setup_remote_docker
       - checkout
+      - <<: *attach_workspace
+      - <<: *load_docker_image
       - run:
-          name: Trigger release candidate builds
-          command: |
-            source ci_utils.sh
-            trigger_artifact_build anchore-cli
+          name: Push to Dockerhub
+          command: make << parameters.make_job >>
 
 workflows:
   default_workflow:
     jobs:
-      - unit_tests_py27
-      - unit_tests_py36
```

Comment:
I see these environments are removed in several places, but I don't see where Python 2.7 is used back again. The idea is that these both need to get tested. Is it me that is missing something about this setup that can't find where 2.7 is invoked?

Reply:
I changed the unit/functional tests to use configurable environments in the job workflow since the individual jobs were identical. This makes for a streamlined config file & allows the jobs to run in parallel.